src/utils/errorSerializer.ts

```diff
@@ -34,60 +38,81 @@ import UserTimeoutError from "../errors/UserTimeoutError";
   [ErrorCode.ERR_WEBEID_CONTEXT_INSECURE]:        ContextInsecureError,
   [ErrorCode.ERR_WEBEID_EXTENSION_UNAVAILABLE]:   ExtensionUnavailableError,
+  [ErrorCode.ERR_WEBEID_MISSING_PARAMETER]:       MissingParameterError,
   [ErrorCode.ERR_WEBEID_NATIVE_INVALID_ARGUMENT]: NativeInvalidArgumentError,
   [ErrorCode.ERR_WEBEID_NATIVE_FATAL]:            NativeFatalError,
   [ErrorCode.ERR_WEBEID_NATIVE_UNAVAILABLE]:      NativeUnavailableError,
   [ErrorCode.ERR_WEBEID_USER_CANCELLED]:          UserCancelledError,
   [ErrorCode.ERR_WEBEID_USER_TIMEOUT]:            UserTimeoutError,
   [ErrorCode.ERR_WEBEID_VERSION_INVALID]:         VersionInvalidError,
   [ErrorCode.ERR_WEBEID_VERSION_MISMATCH]:        VersionMismatchError,
+  [ErrorCode.ERR_WEBEID_UNKNOWN_ERROR]:           UnknownError,
 };
 
-export function serializeError(error: any): any {
-  const {
-    message,
-    name,
-    fileName,
-    lineNumber,
-    columnNumber,
-    stack,
-  } = error;
-
-  return {
-    ...(
-      Object.fromEntries(
-        Object.getOwnPropertyNames(error)
-          .map((prop) => [prop, error[prop]])
-      )
-    ),
-
-    message,
-    name,
-    fileName,
-    lineNumber,
-    columnNumber,
-    stack,
-  };
+export function serializeError(error: Error): SerializedError {
+  if (typeof error != "object" || !(error instanceof Error)) {
+    error = new UnknownError(typeof error == "string" ? error : undefined);
+  }
+
+  const result: Record<string, unknown> = {};
+
+  for (const prop of Object.getOwnPropertyNames(error)) {
+    Reflect.set(result, prop, Reflect.get(error, prop));
+  }
+
+  for (const key of ["name", "stack"]) {
+    const errorAsObject = (error as unknown as Record<string, unknown>);
+    if (!result[key] && errorAsObject?.[key] != null) {
+      Reflect.set(result, key, errorAsObject[key]);
+    }
+  }
+
+  if (!result.code) {
+    Reflect.set(result, "code", ErrorCode.ERR_WEBEID_UNKNOWN_ERROR);
+    Reflect.set(result, "name", UnknownError.name);
+  }
+
+  return result as SerializedError;
 }
 
-export function deserializeError(errorObject: any): any {
-  let error;
+function isSerializedError(obj: unknown): obj is SerializedError {
+  if (typeof obj !== "object" || obj === null) {
+    return false;
+  }
+
+  const errorObject = obj as SerializedError;
+
+  return (
+    typeof errorObject.code    === "string" &&
+    typeof errorObject.message === "string" &&
+
+    errorCodeToErrorClass[errorObject.code] != null
+  );
+}
 
-  if (typeof errorObject.code == "string" && errorObject.code in errorCodeToErrorClass) {
+export function deserializeError(errorObject: SerializedError): InstanceType<KnownErrorConstructors> {
+  let error: InstanceType<KnownErrorConstructors>;
+
+  if (isSerializedError(errorObject)) {
     const CustomError = errorCodeToErrorClass[errorObject.code];
 
     error = new CustomError();
+
+    Object.entries(errorObject).forEach(([key, value]) => {
+      Reflect.set(error, key, value);
```

Comment:
Same direct assignment question here:
```suggestion
      error[key] = value;
```